### PR TITLE
[python] Allow `adata.X` to be `None`

### PR DESCRIPTION
**Pull request by johnkerl**
**Issue and/or context:** Issue #1672. Extends PR #1762.

**Changes:**

**Notes for Reviewer:**

### CODE REVIEW

**Comment by johnkerl** [comment]
This is stacked atop the blocking PR #1766 until that PR is merged, at which point this PR will be on `main`.

**Comment by eddelbuettel** [comment]
@johnkerl In the very first line you refer to _both_ issue _and PR_ as 1762.  Did you mean 1762 and 1763 ?

File: apis/python/src/tiledbsoma/io/ingest.py [hunk 1]
```diff
@@ -510,10 +510,12 @@ def from_anndata(
                 #   chunkwise into memory.
                 # Using the latter allows us to ingest larger .h5ad files without OOMing.
 
-                # Some AnnData objects have no X at all.
+                # Some AnnData objects have no X at all. This might be a missing attribute
+                # anndata.X; it might be anndata.X present, but with value None.
                 has_X = True
                 try:
                     anndata.X
+                    has_X = anndata.X is not None
```
Comment (thetorpedodog) [inline apis/python/src/tiledbsoma/io/ingest.py:518]: I would just say `has_X = anndata.X is not None` and remove the `anndata.X` and `has_X = True` above, like:

```python
# Some AnnData objects ...
try:
    has_X = anndata.X is not None
except ...:
    ...
```

Alternately, you might restructure this to eliminate the `has_X` variable entirely:

```python
try:
    anndata_X = anndata.X
except (...):
    pass
else:
    if anndata_X is not None:
        ...
```

**Review by thetorpedodog — APPROVED** [review]
minor suggestion(s) which you can take or leave

**Review by nguyenv — APPROVED** [review]

**Pull request merged**